pkg/plugins/envvars/envvars.go

```diff
@@ -46,13 +50,21 @@ func (s *Service) Get(_ context.Context, p *plugins.Plugin) []string {
 		hostEnv = append(hostEnv, s.license.Environment()...)
 	}
 
+	if p.ExternalServiceRegistration != nil && s.cfg.Features.IsEnabled(featuremgmt.FlagExternalServiceAuth) {
```

Comment:
Nit. Rather than doing this here I would perhaps do this in the initializer instead and add a new field to `plugins.Plugin` holding the external service registration struct to be used here for populating env vars.

Main reason it feels like the service registration is happening a couple layers down in the call chain so could be hard to reason about this compared to having this closer to loader/initializer where most other things are being initialized/setup.